### ISSUE TITLE
MqttClientWrapper: An error occured while connecting to MQTT broker. Retrying in 5 seconds.

### ISSUE THREAD

**Issue by antonhagg** [yp87/leaf2mqtt#69]
Trying to connect to Beebotte that I use as a broker, but it seems to be an issue. Has anyone else testet this broker?

"INFO: 2023-11-29 14:40:30.605198: MqttClientWrapper: Creating MQTT client with mqtt.beebotte.com:8883 listening on leaf/test."
"INFO: 2023-11-29 14:40:30.607382: MqttClientWrapper: Connecting..."
"WARNING: 2023-11-29 14:40:43.403963: MqttClientWrapper: An error occured while connecting to MQTT broker. Retrying in 5 seconds."

**Comment by antonhagg** [comment]
This is also in the logs.
"INFO: 2023-11-29 15:02:25.734287: MqttClientWrapper: mqtt-client::NoConnectionException: The maximum allowed connection attempts ({3}) were exceeded. The broker is not responding to the connection request message (Missing Connection Acknowledgement?"

"FINEST: 2023-11-29 15:02:25.734327: MqttClientWrapper: #0      SynchronousMqttServerConnectionHandler.internalConnect (package:mqtt_client/src/connectionhandling/server/mqtt_client_synchronous_mqtt_server_connection_handler.dart:116)"

**Comment by antonhagg** [comment]
I have done some testing today and it seems this is related to SSL/TLS that is not supported as far as I can see. 
Would it be possible to add support for SSL. Se here for an example: https://github.com/shamblett/mqtt_client/blob/master/example/mqtt_server_client_secure.dart

**Comment by astarkb** [comment]
@antonhagg I get the same error as you. Did you find a way to make it work?

**Comment by astarkb** [comment]
I was able to fix the problem by creating a new HA user, and putting the credentials of this user into the MQTT_USERNAME and MQTT_PASSWORD configuration fields of leaf2MQTT.

Now I get this error: "An error occured while logging in. Please make sure you have selected the right LEAF_TYPE, LEAF_USERNAME and LEAF_PASSWORD.". From what i could find, this is because i have a Canadian 2023 Leaf which isn't supported by this add-on.. I emailed Nissan to ask them to open their API to Canadians, but I doubt it will ever work..

**Comment by antonhagg** [comment]
Yes, I have a fork that allows for secure connection. But it stopped working yesterday because Nissan changed the API. Will update my fork when the original fork updates .
